### PR TITLE
Add set and iterable base classes.

### PR DESCRIPTION
**Pull request by linus-sherrill**
Also added image container set.

### CODE REVIEW

**Comment by dstoup** [comment]
Jenkins test this please.

File: vital/tests/test_image_container_set.cxx [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+{
+  using namespace kwiver::vital;
+  // Make a vector of empty images.
+  ic_vec_t img_vec = make_simple_ic_vec();
+  simple_image_container_set sics( img_vec );
+
+  // We should be able to iterate 3 times with img_vec and the returns should
+  // be equivalent to the direct vector iterator
+  ic_vec_t::iterator                   vec_it = img_vec.begin();
+  simple_image_container_set::iterator sic_it = sics.begin();
```
Comment (dstoup) [inline vital/tests/test_image_container_set.cxx]: Curious why we're using being() to start and cend() below. Shouldn't those be consistent depending on whether the iterator is const? Probably it doesn't matter in this context.

**Comment by dstoup** [comment]
Looks like the mac test failures are directly related to this branch.

**Comment by linus-sherrill** [comment]
jenkins, test this please

**Comment by dstoup** [comment]
The Windows build failed because of my machine being down. I'll rerun it quickly to sanity check the last commit.

**Comment by dstoup** [comment]
Jenkins test this please

**Pull request merged**